frontend/src/StoryDisplays/Stories.jsx

```diff
@@ -1,25 +1,147 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef, useCallback } from 'react';
+import StoriesList from './StoriesList';
+import StoryViewer from './StoryViewer';
 import './Stories.css';
 
-const Stories = () => {
-    const [stories, setStories] = useState([]);
-
-    useEffect(() => {
-        fetch('http://localhost:3000/api/stories')
-            .then(response => response.json())
-            .then(data => setStories(data))
-            .catch(error => console.error('Error fetching stories:', error));
-    }, []);
-
-    return (
-        <div className="stories-container">
-            {stories.map(story => (
-                <div key={story.id} className="story-circle">
-                    <img src={story.fileUrl} alt="Story" className="story-image" />
-                </div>
-            ))}
+const Stories = ({ currentUser }) => {
+  const [stories, setStories] = useState([]);
+  const storiesListRef = useRef(new StoriesList());
+  const [viewingStories, setViewingStories] = useState(null);
+
+  const fetchStories = useCallback(() => {
+    fetch('http://localhost:3000/api/stories')
+      .then(response => response.json())
+      .then(data => {
+        const storiesList = storiesListRef.current;
+        storiesList.removeExpiredStories();
```

Comment:
Oh I see, you remove the expired on stories on client side. This is fine too, although from a system design perspective we could also consider shifting this logic to the backend so the client can use the data immediately without having to perform additional validity checks.